scrapy/dupefilters.py

```diff
@@ -40,6 +40,10 @@ def __init__(self, path=None, debug=False):
             self.fingerprints.update(x.rstrip() for x in self.file)
 
     @classmethod
+    def from_crawler(cls, crawler):
+        return cls.from_settings(crawler.settings)
```

Comment:
why is this method needed?

Reply:
Hey @kmike! Thanks for the quick response.
First I thought about leaving only `from_crawler`, but to keep this backwards compatible (and also because of your comment in the original issue: "add from_crawler support to dupefilters _in addition_ to from_settings") I kept both methods. It follows the same chain of precedence that can be found in https://github.com/scrapy/scrapy/blob/1.4/scrapy/middleware.py#L35-L40

Reply:
It makes sense to follow the same logic, but if you remove from_crawler method from this class, it would do exactly the same (i.e. from_crawler doesn't seem to be needed here). 

Also, if you subclass from a dupefilter and override from_setting, overridden method won't be called after this PR.

Reply:
You're right, I see your point now, thanks! I'll remove it :+1:

Reply:
@kmike: Related to this: is it possible that the `MiddlewareManager` is affected by this too? If I'm not mistaken, all of its subclasses (DownloaderMiddlewareManager, SpiderMiddlewareManager, ExtensionManager, ItemPipelineManager) are created using `from_crawler`, and none of them are supposed to be overridden by the user (except maybe the pipeline manager using the `ITEM_PROCESSOR` setting, but that it's not even documented). The only occurrence of the manager using `from_settings` is in the `tests/test_middleware.py` file (`test_enabled_from_settings`)
What do you think about some cleanup in https://github.com/scrapy/scrapy/blob/1.4/scrapy/middleware.py#L28-L58 to keep only the `from_crawler` method? In a different PR, of course.